test/Parse/subscripting.swift

```diff
@@ -80,9 +80,8 @@ let y2 = Y2() // expected-note{{change 'let' to 'var' to make it mutable}}{{1-4=
 _ = y2[0] // expected-error{{cannot use mutating getter on immutable value: 'y2' is a 'let' constant}}
 
 // Parsing errors
-// FIXME: Recovery here is horrible
 struct A0 {
-  subscript // expected-error{{expected '(' for subscript parameters}}
+  subscript // expected-error {{subscripts cannot have a name}} {{5-7=}}
     i : Int
```

Comment:
This is interesting; because of the following colon, we ought to realize that it's parens that are missing, not a spurious name being added. I'm not sure this is a net improvement in this case.

Mind checking for a colon too? That would help initializers as well, though less so since `init foo bar : Int` could be a mistake for `init foo(bar: Int)` or `init(foo bar: Int)`.